models/users.js

```diff
@@ -58,10 +58,17 @@ const addOrUpdate = async (userData, userId = null) => {
  */
 const fetchUsers = async (query) => {
   try {
-    const snapshot = await userModel
+    let snapshot = await userModel
```

Comment:
So, if `query.role === "members"`, we should still be making this call for no reason, correct?